jupyterhub/apihandlers/users.py

```diff
@@ -18,6 +18,47 @@ def get(self):
         users = self.db.query(orm.User)
         data = [ self.user_model(u) for u in users ]
         self.write(json.dumps(data))
+    
+    @admin_only
+    @gen.coroutine
+    def post(self):
+        data = self.get_json_body()
+        if not data or not isinstance(data, dict) or not data.get('usernames'):
+            raise web.HTTPError(400, "Must specify at least one user to create")
+        
+        usernames = data.pop('usernames')
+        admin = data.get('admin', False)
+        self._check_user_model(data)
+        
+        to_create = []
+        for name in usernames:
+            user = self.find_user(name)
+            if user is not None:
+                self.log.warn("User %s already exists" % name)
+            else:
+                to_create.append(name)
+        
+        if not to_create:
+            raise web.HTTPError(400, "All %i users already exist" % len(usernames))
+        
+        created = []
+        for name in to_create:
+            user = self.user_from_username(name)
+            if admin:
+                user.admin = True
+                self.db.commit()
```

Comment:
I'm not sure I understand the question. Modifying the object and sending the changes to the database are two separate actions. You don't want every change to hit the database immediately because that can get super slow.